calico_node/startup/startup.go

```diff
@@ -781,83 +855,154 @@ func ensureDefaultConfig(cfg *api.CalicoAPIConfig, c *client.Client, node *api.N
+			newBGPCfg.Name = GLOBAL_BGP_CONFIG_NAME
+			newBGPCfg.Spec.NodeToNodeMeshEnabled = &DEFAULT_NODE_TO_NODE_MESH
+			newBGPCfg.Spec.ASNumber = &DEFAULT_ASNUMBER
+			newBGPCfg.Spec.LogSeverityScreen = DEFAULT_LOG_SEVERITY
+			_, err = c.BGPConfigurations().Create(ctx, newBGPCfg, options.SetOptions{})
+			if err != nil {
+				log.WithError(err).WithField("BGPConfig", newBGPCfg).Errorf("Error creating BGP config")
+				return err
+			}
+		} else {
+			log.WithError(err).WithField("BGPConfig", GLOBAL_BGP_CONFIG_NAME).Errorf("Error getting BGP config")
+			return err
+		}
 	} else {
-		log.WithField(key, val).Debug("Global Felix value already assigned")
-		return nil
-	}
-}
+		updateNeeded := false
+		if bgpConfig.Spec.NodeToNodeMeshEnabled == nil {
+			bgpConfig.Spec.NodeToNodeMeshEnabled = &DEFAULT_NODE_TO_NODE_MESH
+			updateNeeded = true
+		} else {
+			log.WithField("NodeToNodeMeshEnabled", bgpConfig.Spec.NodeToNodeMeshEnabled).Debug("Global BGP value already assigned")
+		}
 
-// ensureNodeFelixConfig ensures that the supplied felix config value
-// is initialized, and if not initialize it with the supplied default.
-func ensureNodeFelixConfig(c *client.Client, host, key, def string) error {
-	if val, assigned, err := c.Config().GetFelixConfig(key, host); err != nil {
-		return err
-	} else if !assigned {
-		return c.Config().SetFelixConfig(key, host, def)
-	} else {
-		log.WithField(key, val).Debug("Host Felix value already assigned")
-		return nil
+		if bgpConfig.Spec.ASNumber == nil {
+			bgpConfig.Spec.ASNumber = &DEFAULT_ASNUMBER
+			updateNeeded = true
+		} else {
+			log.WithField("DefaultNodeASNumber", bgpConfig.Spec.ASNumber).Debug("Global BGP value already assigned")
+		}
+
+		if bgpConfig.Spec.LogSeverityScreen == "" {
+			bgpConfig.Spec.LogSeverityScreen = DEFAULT_LOG_SEVERITY
+			updateNeeded = true
+		} else {
+			log.WithField("LogSeverityScreen", bgpConfig.Spec.LogSeverityScreen).Debug("Global BGP value already assigned")
+		}
+
+		if updateNeeded {
+			_, err = c.BGPConfigurations().Update(ctx, bgpConfig, options.SetOptions{})
+			if err = checkConflictError(err); err != nil {
+				log.WithError(err).WithField("BGPConfig", bgpConfig).Errorf("Error updating BGP config")
+				return err
+			}
+		}
 	}
+
+	return nil
 }
 
 // updateClusterType ensures that the ClusterType in the Felix global
 // config has the data contained in the passed in clusterType.  This includes
 // merging what was already set in ClusterType with the passed in clusterType.
-func updateClusterType(c *client.Client, key, clusterType string) error {
+func updateClusterType(ctx context.Context, c client.Interface, clusterType string) error {
 	// If no data is passed in to add, then nothing to do.
 	if clusterType == "" {
 		return nil
 	}
-	global := ""
-	if val, assigned, err := c.Config().GetFelixConfig("ClusterType", global); err != nil {
+
+	// Default ClusterInfo resource should already have been created by this time,
```

Comment:
To minimize the number of queries we do on startup we should also be handling the cluster GUID here too since it is in the same resource.